ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+use constant LINE_VALUE_AS_IS => 0;
+use constant LINE_VALUE_BOOLEAN => 1;
+use constant LINE_VALUE_HOST_LIST => 2;
+use constant LINE_VALUE_INSTANCE_PARAMS => 3;
+use constant LINE_VALUE_ARRAY => 4;
+use constant LINE_VALUE_HASH_KEYS => 5;
+use constant LINE_VALUE_STRING_HASH => 6;
+use constant LINE_VALUE_OPT_NONE => 0;
+use constant LINE_VALUE_OPT_SINGLE => 1;
+use constant LINE_FORMAT_DEFAULT => LINE_FORMAT_PARAM;
+use constant LINE_QUATTOR_COMMENT => "\t\t# Line generated by Quattor";
+use constant LINE_OPT_DEF_REMOVE_IF_UNDEF => 0;
+use constant LINE_OPT_DEF_ALWAYS_RULES_ONLY => 0;
+
+
+# Define and export the constants used to define rules
+# Keep in sync with ncm-xrootd
+# See http://www.perlmonks.org/?node_id=1072731 for the exporting method
+#use warnings;
+#my %rule_constants;
+#BEGIN {
+#  %rule_constants = (
+#    "LINE_FORMAT_PARAM" => 1,
+#    "LINE_FORMAT_ENVVAR" => 2,
+#    "LINE_FORMAT_XRDCFG" => 3,
+#    "LINE_FORMAT_XRDCFG_SETENV" => 4,
+#    "LINE_FORMAT_XRDCFG_SET" => 5,
+#    "LINE_VALUE_AS_IS" => 0,
+#    "LINE_VALUE_BOOLEAN" => 1,
+#    "LINE_VALUE_HOST_LIST" => 2,
+#    "LINE_VALUE_INSTANCE_PARAMS" => 3,
+#    "LINE_VALUE_ARRAY" => 4,
+#    "LINE_VALUE_HASH_KEYS" => 5,
+#    "LINE_VALUE_STRING_HASH" => 6,
+#    "LINE_VALUE_OPT_NONE" => 0,
+#    "LINE_VALUE_OPT_SINGLE" => 1,
+#    "LINE_FORMAT_DEFAULT" => LINE_FORMAT_PARAM,
+#    "LINE_QUATTOR_COMMENT" => "\t\t# Line generated by Quattor",
+#    "LINE_OPT_DEF_REMOVE_IF_UNDEF" => 0,
+#    "LINE_OPT_DEF_ALWAYS_RULES_ONLY" => 0,
+#  );
+#}
+
+#use constant \%rule_constants;
+
+use base 'Exporter';
+
+my @rule_constants = ('LINE_FORMAT_PARAM',
+                   'LINE_FORMAT_ENVVAR',
+                   'LINE_FORMAT_XRDCFG',
+                   'LINE_FORMAT_XRDCFG_SETENV',
+                   'LINE_FORMAT_XRDCFG_SET',
+                   'LINE_VALUE_AS_IS',
+                   'LINE_VALUE_BOOLEAN',
+                   'LINE_VALUE_HOST_LIST',
+                   'LINE_VALUE_INSTANCE_PARAMS',
+                   'LINE_VALUE_ARRAY',
+                   'LINE_VALUE_HASH_KEYS',
+                   'LINE_VALUE_STRING_HASH',
+                   'LINE_VALUE_OPT_NONE',
+                   'LINE_VALUE_OPT_SINGLE',
+                   'LINE_FORMAT_DEFAULT',
+                   'LINE_QUATTOR_COMMENT',
+                   'LINE_OPT_DEF_REMOVE_IF_UNDEF',
+                   'LINE_OPT_DEF_ALWAYS_RULES_ONLY',
+                  );
+our @EXPORT_OK;
+our %EXPORT_TAGS;
+push @EXPORT_OK, @rule_constants;
+$EXPORT_TAGS{rule_constants} = \@rule_constants;
+
+
+# Backup file extension
+use constant BACKUP_FILE_EXT => ".old";
+
+
+
+# This function formats an attribute value based on the value format specified.
+#
+# Arguments :
+#        attr_value : attribue value
+#        line_fmt : line format (see LINE_FORMAT_xxx constants)
+#        value_fmt : value format (see LINE_VALUE_xxx constants)
+sub formatAttributeValue () {
+  my $function_name = "formatAttributeValue";
```

Comment:
The whole file is based on 2 space indent... I'm reluctant to change it right now...